### ISSUE TITLE
Error: ``Cannot colocate nodes...''

### ISSUE THREAD

**Issue by yuchenlin** [HKUST-KnowComp/R-Net#7]
Hi Wenxuan,

Many thanks for this wonderful and educational implementation. 
I have encountered an error while trying ```python config.py --mode debug```.

>InvalidArgumentError (see above for traceback): Cannot colocate nodes 'global_norm/L2Loss_32' and 'gradients/match/bw/CudnnRNN_grad/CudnnRNNBackprop' because no device type supports both of those nodes and the other nodes colocated with them.
Colocation Debug Info:
Colocation group had the following types and devices:
CudnnRNNBackprop: GPU
Identity:
L2Loss: CPU
         [[Node: global_norm/L2Loss_32 = L2Loss[T=DT_FLOAT, _class=["loc:@gradients/match/bw/CudnnRNN_grad/CudnnRNNBackprop"]](gradients/match/bw/CudnnRNN_grad/tuple/control_dependency_3)]]

  All the configs are by default.

**Comment by yuchenlin** [comment]
It works well when I set ```use_cudnn``` as False. (~~1.2it/s without Bucket)
However, it could be great if there are some ideas why it could not work.

**Comment by xiongyifan** [comment]
If use_cudnn is True. The training step will use GPU to train. But you used CPU to train.

**Comment by yuchenlin** [comment]
@xiongyifan thanks. However, i can train with GPU after setting it as False.  That’s also confusing.

**Comment by xiongyifan** [comment]
@yuchenlin ... ... I am also confusing. Maybe because of the version. When I was running [a RNN example](https://www.tensorflow.org/tutorials/recurrent) that uses the CudnnLSTM, I could not use Adam to optimize the CudnnLSTM. But in this project, we can use Adam to optimize CudnnGRU. I am not sure ...  I am confusing ...

**Comment by yuchenlin** [comment]
Maybe also due to I am training on windows server or my cudnn version, although many other TensorFlow models can run in my environment without problems. Anyway, it can work well now and the speed is quite okay. Thanks very much! I guess we can close this issue for it may be my own problem.

**Comment by xiongyifan** [comment]
@yuchenlin If you find the problem. Please tell me.

**Comment by liuny05** [comment]
I have same issue on windows10, and on ubuntu16.04 it works well.

**Comment by yoonforh** [comment]
i think i have similar problem when i call **tf.clip_by_global_norm(gradients, max_norm)** with cudnn lstm

**Comment by mickyyu96** [comment]
@yuchenlin how you solve the problem ? i got the same issue
